packages/console/src/pages/OrganizationRoleDetails/Permissions/index.tsx

```diff
@@ -1,77 +1,92 @@
 import { useTranslation } from 'react-i18next';
-import useSWR from 'swr';
 
 import Plus from '@/assets/icons/plus.svg';
 import ActionsButton from '@/components/ActionsButton';
 import Breakable from '@/components/Breakable';
+import EditScopeModal, { type EditScopeData } from '@/components/EditScopeModal';
 import EmptyDataPlaceholder from '@/components/EmptyDataPlaceholder';
-import ManageOrganizationPermissionModal from '@/components/ManageOrganizationPermissionModal';
 import Button from '@/ds-components/Button';
 import DynamicT from '@/ds-components/DynamicT';
 import Search from '@/ds-components/Search';
 import Table from '@/ds-components/Table';
 import Tag from '@/ds-components/Tag';
-import useApi, { type RequestError } from '@/hooks/use-api';
+import useApi from '@/hooks/use-api';
 import useSearchParametersWatcher from '@/hooks/use-search-parameters-watcher';
 
+import ResourceName from './ResourceName';
 import * as styles from './index.module.scss';
+import useOrganizationRoleScopes from './use-organization-role-scopes';
 
-const organizationRolesPath = 'api/organization-roles';
+type OrganizationRoleScope = OrganizationScope | Scope;
+
+const isResourceScope = (scope: OrganizationRoleScope): scope is Scope => 'resourceId' in scope;
 
 type Props = {
   organizationRoleId: string;
 };
 
 function Permissions({ organizationRoleId }: Props) {
+  const organizationRolePath = `api/organization-roles/${organizationRoleId}`;
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
   const api = useApi();
 
-  const { data, error, isLoading, mutate } = useSWR<OrganizationScope[], RequestError>(
-    `${organizationRolesPath}/${organizationRoleId}/scopes`
-  );
+  const { organizationScopes, resourceScopes, error, isLoading, mutate } =
+    useOrganizationRoleScopes(organizationRoleId);
 
   const [{ keyword }, updateSearchParameters] = useSearchParametersWatcher({
     keyword: '',
   });
 
-  const filteredData = useMemo(() => {
-    if (keyword) {
-      return data?.filter((roleScope) => roleScope.name.includes(keyword));
-    }
-    return data;
-  }, [data, keyword]);
-
-  const [editPermission, setEditPermission] = useState<OrganizationScope>();
-
-  const scopeRemoveHandler = useCallback(
-    (scopeToRemove: OrganizationScope) => async () => {
-      await api.put(`${organizationRolesPath}/${organizationRoleId}/scopes`, {
-        json: {
-          organizationScopeIds:
-            data?.filter((scope) => scope.id !== scopeToRemove.id).map(({ id }) => id) ?? [],
-        },
-      });
+  const filterScopes = useCallback(
+    (scopes: OrganizationRoleScope[]) => scopes.filter(({ name }) => name.includes(keyword)),
+    [keyword]
+  );
+
+  const filteredScopes = useMemo(
+    () =>
+      keyword
+        ? [...filterScopes(resourceScopes), ...filterScopes(organizationScopes)]
+        : [...resourceScopes, ...organizationScopes],
+    [filterScopes, keyword, organizationScopes, resourceScopes]
+  );
```

Comment:
nit: `filterScopes` and `filteredScopes` are a little bit confusing... Consider renaming or combing these two hooks

Reply:
Updated.